src/state/contexts/TransactionState.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  Dispatch,
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+} from 'react';
+
+import {
+  AntInteraction,
+  CONTRACT_TYPES,
+  ContractType,
+  REGISTRY_INTERACTION_TYPES,
+  RegistryInteraction,
+} from '../../types';
+import { TransactionAction } from '../reducers/TransactionReducer';
+
+export type TransactionState = {
+  transactionData: { [x: string]: any }; // data that will be used to perform the transaction.
+  contractType: ContractType;
+  interactionType: AntInteraction | RegistryInteraction;
+  workflowStage: number;
+};
+
+export type TransactionStateProviderProps = {
+  reducer: React.Reducer<TransactionState, TransactionAction>;
+  children: React.ReactNode;
+};
+
+export const initialTransactionState: TransactionState = {
+  transactionData: {},
+  contractType: CONTRACT_TYPES.REGISTRY,
+  interactionType: REGISTRY_INTERACTION_TYPES.BUY_RECORD,
+  workflowStage: 1,
```

Comment:
to decouple from the workflow component, what if this is instead `validate | pending | deployed | confirmed` or something similar

Reply:
that is, numbers are unclear in the context of this provider

Reply:
changed